core/controllers/resources.py

```diff
@@ -40,79 +40,40 @@ def get(self, generator_id):
             raise self.PageNotFoundException
 
 
-class ImageDevHandler(base.BaseHandler):
-    """Handles image retrievals (only in dev -- in production, image files are
-    served from GCS).
+class AssetDevHandler(base.BaseHandler):
+    """Handles image and audio retrievals (only in dev -- in production,
+    image and audio files are served from GCS).
     """
 
-    _IMAGE_PATH_PREFIX = 'image'
-
     @acl_decorators.open_access
-    def get(self, exploration_id, encoded_filename):
-        """Returns an image.
+    def get(self, exploration_id, asset_type, encoded_filename):
+        """Returns an asset file.
 
         Args:
             exploration_id: the id of the exploration.
-            encoded_filename: a string representing the image filename. This
+            asset_type: type of the asset, either image or audio
+            encoded_filename: a string representing the asset filename. This
               string is encoded in the frontend using encodeURIComponent().
         """
         if not constants.DEV_MODE:
```

Comment:
Validate the asset_type arg first and raise exception if it's not what you expect.